plugins/modules/nd_pcv.py

```diff
@@ -213,15 +213,17 @@ def main():
         if file:
             if not os.path.exists(file):
                 nd.fail_json(msg="File not found : {0}".format(file))
-            # check whether file content is a valid json
-            if ndi.is_json(open(file, "rb").read()) is False:
-                extract_data = ndi.load(open(file))
-            else:
-                extract_data = json.loads(open(file, "rb").read())
-            if isinstance(extract_data, list):
-                ndi.cmap = {}
-                tree = ndi.construct_tree(extract_data)
-                ndi.create_structured_data(tree, file)
+            # Check if file extension is XML. If XML, skip JSON validation/transformation
+            if not file.endswith(".xml"):
```

Comment:
Should we base the content validation on the file extension?
I think we could achieve a nicer user experience by catching the JSON exception and checking if it is an actual XML document.

What do you think?

Reply:
It's another option, indeed. I am not sure how common it is for customers to use an XML file without the proper extension, but of course that would be a more solid experience. Any ideas how to check if it's a valid XML file, other than checking the content for XML tags?

Reply:
Think there are more ways of achieving this but I would say either lxml (https://lxml.de/validation.html) or xmlschema (https://xmlschema.readthedocs.io/en/latest/usage.html#validation)